### DISCUSSION TITLE
SRP table empty when turn off and turn on TBR (2 nano leaf bulbs connected and showing in otbr-web) but SRP records empty

### DISCUSSION THREAD

**Discussion by gadekula** [openthread/openthread#10682] [Q&A]
In continuation to #10669 observing that the SRP server table is empty For example, with the srp server service command
Yes as mentioned in title we are unable to see SRP update when TBR turned on back
Help us to figure it out, how we can fastly get SRP record updated from nano leaf bulbs

**Comment by jwhui** [comment]
Are you able to reproduce the same problem using the OpenThread SRP Client?

https://github.com/openthread/openthread/blob/main/src/cli/README_SRP.md#start-srp-client

This would help identify whether there is an issue specific to the Nanoleaf bulb.

**Reply by gadekula** [reply]
Thanks for the reply 
Sure will check that part
Any chance that we can explicitly query SRP client (nano leaf bulbs (SRP client) which is not updating fastly) from TBR
Note : As nano leaf bulbs is end product we cannot debug from SRP client

**Comment by gadekula** [comment]
Hi @jwhui 
Reg OpenThread SRP Client:
set up : 2 TBR's used, 1 as SRP server and 1 as SRP client. both connected in same ip N/W.

copying logs as unable to attach as file
**SRP client logs**
` ot-ctl
srp client help
autostart
callback
host
keyleaseinterval
leaseinterval
server
service
start
state
stop
ttl
Done
>
> dataset networkkey d8adae16404a0a26959fddbae62be38b
Done
> dataset commit active
Done
> ifconfig up
Done
> thread start
Done
> state
detached
Done
> state
router
Done
> ipaddr
fd76:6d12:b168:6716:0:ff:fe00:e400
fda0:24e2:4acc:1:b5af:a0ee:f79f:5681
fd76:6d12:b168:6716:726:8123:2f3a:3ba8
fe80:0:0:0:44ab:9c39:1325:855d
Done
> srp client host name my-host
Done
> srp client host address fd76:6d12:b168:6716:726:8123:2f3a:3ba8
Done
> srp client service add my-service _ipps._tcp 12345
Done
> srp client autostart enable
Done
> srp client host
name:"my-host", state:Registered, addrs:[fd76:6d12:b168:6716:726:8123:2f3a:3ba8]
Done
> srp client service
instance:"my-service", name:"_ipps._tcp", state:Registered, port:12345, priority:0, weight:0
Done
>
`
**SRP server logs:**

root@LGwebOSTV:~# ot-ctl
srp server help
addrmode
auto
disable
domain
enable
host
lease
seqnum
service
state
ttl
Done
> dataset init new
Done
> dataset
Active Timestamp: 1
Channel: 25
Channel Mask: 0x07fff800
Ext PAN ID: f56da5e0b8bb4e4e
Mesh Local Prefix: fd76:6d12:b168:6716::/64
Network Key: d8adae16404a0a26959fddbae62be38b
Network Name: OpenThread-9fdd
PAN ID: 0x9fdd
PSKc: 901c20642f1c460495b621f3823a1eca
Security Policy: 672 onrc 0
Done
> dataset commit active
Done
> ifconfig up
Done
> thread start
Done
> state
leader
Done
> ipaddr
fd76:6d12:b168:6716:0:ff:fe00:fc11
fd76:6d12:b168:6716:0:ff:fe00:fc10
fd76:6d12:b168:6716:0:ff:fe00:fc38
fda0:24e2:4acc:1:71c0:feaf:8989:9a55
fd76:6d12:b168:6716:0:ff:fe00:fc00
fd76:6d12:b168:6716:0:ff:fe00:f400
fd76:6d12:b168:6716:1e88:fb6:40bf:5aea
fe80:0:0:0:8827:6c53:644b:860c
Done
> srp server enable
Done
> **srp server host
Done**
> **srp server service
Done**
>

Unable to see service or host in SRP server
we are also checking from our end, why SRP records not showing in server
Please let me know anything we are missing.

**Reply by abtink** [reply]
Suggestions for things to check:

- The SRP client output shows that it has successfully registered with a server.
- Do you happen to have any other BR in your setup (part of the same Thread mesh)?

**Comment by gadekula** [comment]
@abtink as mentioned earlier we used 2 TBR's
Both are in same thread network.
Eg:
TBR:1 as SRP server 
TBR:2 as SRP client 
Please let me know if any info required

**Reply by jwhui** [reply]
Can you show the IPv6 address of the SRP Server that the SRP Client is using?

```
srp client server
```

**Reply by abtink** [reply]
@gadekula, then check the `srp server host` on both TBRs, particularly on `TBR 2`.  It's likely the services are registered there.

Also, for related discussions, suggest using the "Write a reply" box under previous comments to keep everything organized, instead of starting a new entry using "Suggest an Answer".

**Reply by gadekula** [reply]
@abtink, thanks for the comments

Another observation, we found
2 TBR's and 2 nano leaf bulbs in same thread network,
SRP server data only available in TBR-1 at a time and if we turn off TBR 1 again SRP registration from nano leaf bulb takes ~4 min or even more time in TBR-2,
Can we see SRP details in both TBR's server service at the same time?

**Reply by abtink** [reply]
> SRP server data is only available on one TBR at a time.

This is the expected behavior. A client registers with a single server, and the TBR functions as an Advertising Proxy, advertising the services via mDNS on the infrastructure link. Service discovery can be performed through mDNS regardless of which server each client chooses. 

For your information, there are ongoing efforts towards  "SRP replication" feature to enable SRP servers to synchronize registered information. This is mainly intended as a backup mechanism. This feature is not currently enabled on `ot-br-posix`. Regardless, service discovery should always be performed through `mDNS` on the infra-link, rather than directly using the SRP server entries.

**Reply by gadekula** [reply]
@abtink Thanks for your detailed info, reg srp server host, will update once I captured info

**Comment by vasukanster** [comment]
Can we use srp client to register 2 different hosts or service records to srp server in same machine? I mean srp client can run 2 times with different hostnames in same machine

**Reply by jwhui** [reply]
OpenThread's SRP Client can only register a single hostname with the SRP Server at a time.

**Reply by vasukanster** [reply]
In general even though hostname can be single, we can create multiple hostnames with /etc/hosts and reverse lookup can be possible with different hostname and ip for dns records. Why it is not allowed in srp client, any reason?

**Reply by jwhui** [reply]
Because it adds complexity and overhead to the implementation and operation. As stated in [draft-ietf-dnssd-srp](https://datatracker.ietf.org/doc/html/draft-ietf-dnssd-srp):

> There is never more than one hostname in a single SRP update. The
> hostname has one or more address RRs (AAAA or A) and a KEY RR
> (used for FCFS naming).

Do you have a strong use case for needing support for registering multiple hostnames simultaneously?

**Reply by vasukanster** [reply]
We are doing some experiments for learning purpose, so we thought this could be possible. We don't have any concrete use case. But I thought below scenarios can be possible
1) iot devices with rich capabilities can host services on behalf of some other iot devices
2) iot devices can publish multiple services with different hostnames that may create easy configuration management
